### PR TITLE
Refactor Kraken2 to use viral db for tests

### PR DESCRIPTION
**Pull request by drpatelh**
# nf-core/viralrecon pull request

Many thanks for contributing to nf-core/viralrecon!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/viralrecon branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/viralrecon)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/viralrecon/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by drpatelh** [comment]
Im hoping the tests pass now that I changed the code to not download the human Kraken2 database...

**Comment by drpatelh** [comment]
PS: I have commented out the `PlasmidID` processes because they are failing and the error was too cryptic for me to resolve at this time of night! Definitely :bed: now. See you in the morning, bright and early :sun_behind_small_cloud:

File: docs/usage.md [hunk 1]
```diff
@@ -284,18 +294,18 @@ Full path to Kraken2 database built from viral genome.
 
 Name for host genome as recognised by Kraken2 when using the `kraken2 build` command as configured on `genomes.config`. Default: 'viral'.
 
-### `--host_kraken2_db`
-
-Full path to Kraken2 database built from host genome.
-
-### `--host_kraken2_name`
+### `--kraken2_use_host_db`
```
Comment (heuermh) [inline docs/usage.md:297]: What do you think of an opposite `--kraken2_use_viral_db` flag, one of which defaults to true?

Or alternatively, `--kraken2_db : { 'host', 'viral' }`?

Or is this flag necessary at all, can we just check `host_kraken2_db` and `viral_kraken2_db`?

File: main.nf [hunk 2]
```diff
@@ -1216,19 +1213,19 @@ process BCFTOOLS_CONSENSUS {
 process SPADES {
     tag "$sample"
     label 'process_medium'
-    publishDir "${params.outdir}/spades", mode: params.publish_dir_mode
+    publishDir "${params.outdir}/denovo/spades", mode: params.publish_dir_mode
```
Comment (heuermh) [inline main.nf:1216]: denovo &rarr; assembly ?

File: docs/usage.md (hunk 1 above)
Comment (drpatelh) [reply docs/usage.md:297]: Yes, I am not 100% happy with the implementation for Kraken. Will refine this all before the release 👍

File: main.nf (hunk 2 above)
Comment (drpatelh) [reply main.nf]: Agreed 👍 https://github.com/nf-core/viralrecon/pull/30/commits/a1d2436004e2680397a02d4dc43ce7b437e100fd

**Pull request merged**